### PR TITLE
Merge `:flogger-system-backend` into `:logging-backend`

### PR DESCRIPTION
**Pull request by yevhenii-nadtochii**
This PR initiates merging of Flogger code into our own modules.

It merges `:flogger-system-backend` into `:logging-backend`. 

In `:flogger-system-backend`, tests have already been migrated to Kotlin and have been moved "as is". Main sources have been moved with updated copyright notices. In the updated files, links to original Flogger sources have been left.

---

Before the merge, these modules were responsible for the following:

| Module | Short description |
|---|---|
| `:flogger-system-backend` | Flogger's default backend (JUL-based) and `DefaultPlatform`. |
| `:logging-backend` | Spine's JUL-based backend that partially delegates to Flogger's default one. |

So, `:logging-backend` provided a backend implementation that was picked up by the platform when discovered on the classpath. From the standpoint of Flogger, it was just **another** backend implementation. But from the standpoint of `spine-logging`, it is our **default** backend implementation **for JVM**. 

Under rare circumstances, It led to Flogger's default implementation being used instead of `spine-logging`'s one. It was possible when no `:logging-backend` is provided, but the classpath has `flogger-system-backend`. In such case, the `DefaultPlatform` rolled back to its default backend. 

Such happened in one of our `tests` when I added assertions about the currently picked up backend and context. It could silently use a default backend from Flogger, which was put on the classpath as a transitive dependency from `Spine.testlib`, which, in turn, was put on the classpath by Gradle configuration of our `jvm-module`.

---

After the merge, `:logging-backend` has the following:

1. Spine's default, JUL-based backend. As it was before the merge.
2. `DefaultPlatform` that now rolls back to Spine's default backend, when there's no any other backend implementation on the classpath.
3. Flogger's default backend that should not be used anymore. We may try to remove it during code migration to Kotlin.

Also, for two suites from our `tests` (that check different combinations of backends with contexts), I've added assertions about the currently used backend and context. It is sort of double check. May be propagated to other tests.

---

### Rename `:logging-backend` to `:default-jvm-backend`

I think this name would better describe the module. And put all backends in `backends` directory.

It can even be split into two:  `jvm-jul-backend` and `jvm-default-backend`. The latter will just be a proxy for the JUL.

A similar approach can be applied to other injectable services: `ContextDataProvider`, `Clock`.

### Applicability for KMP applications

The library is easily applicable to **KMP libraries**. Just add `spine-logging` to `commonMain` dependencies. Though, it would work only for JVM target. 

But for **KMP applications** (which would actually log somewhere), we can't put `spine-logging-backend` to `commonMain`. Only to `jvmMain`.

The case is the following: a user would like to put both `spine-loggin` and `spine-logging-backend` to `commonMain`, and it is expected that in each platform a default logging backend will be picked up.

### Java service definitions

We should use the same approach to declare Java services. Some of them are now declared with a file in META-INF, while others with `@AutoService` annotation. It is better to stick to a single approach within the library.

Also, I've removed comments about Kapt/KSP compatibility with AutoValue. I would create an issue in GitHub for that, rather than document each usage of Kapt.

### Disabled Java linters

Flogger code doesn't match rules declared for Spine modules. As we merge Flogger code into our modules, Java linters begin to fail builds. 

I've disabled Java linters for `:logging-backend`. And it seems safe because our code is in Kotlin.

And as for now, I don't see any better ways to overcome this.

---

Other changes are the following:

1. `Spine.testlib` has been removed from the set of default dependencies of our `jvm-module`. It was bringing `flogger` through transitives. Also, I've removed it from `:fixtures` module, it was unused there. It remains in `:logging` module, but with all `com.google.flogger` artifacts being filtered out.
2. `MetadataSubject.containsEntries(...)` has been removed. It was unused.

---

Update 22.09.2023

The following issues have been created from this PR:

1. [Choose a better name for `:logging-backend`](https://github.com/SpineEventEngine/logging/issues/52).
2. [Remove deprecated `io.spine.logging.Logging` interface](https://github.com/SpineEventEngine/logging/issues/51).
3. [Consider making `:logging-backend` a multiplatform module](https://github.com/SpineEventEngine/logging/issues/53).
4. [Choose a single approach for declaring Java services](https://github.com/SpineEventEngine/logging/issues/54).
5. [Replace Kapt with KSP for processing `@AutoValue` annotations in Kotlin](https://github.com/SpineEventEngine/logging/issues/55).
6. [Enable back Java linters](https://github.com/SpineEventEngine/logging/issues/56).
7. [Assert the actually used backend and context implementations in `tests` modules](https://github.com/SpineEventEngine/logging/issues/57)

### CODE REVIEW

**Comment by yevhenii-nadtochii** [comment]
@armiol @alexander-yevsyukov PTAL

File: flogger/flogger-log4j2-backend/src/main/java/com/google/common/flogger/backend/log4j2/Log4j2BackendFactory.java [hunk 1]
```diff
@@ -17,14 +17,14 @@
...
-import com.google.common.flogger.backend.system.BackendFactory;
+import io.spine.logging.backend.system.BackendFactory;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.core.Logger;
 
 /**
  * BackendFactory for log4j2.
  *
- * <p>When using Flogger's {@link com.google.common.flogger.backend.system.DefaultPlatform}, this
+ * <p>When using Flogger's {@link io.spine.logging.backend.system.DefaultPlatform}, this
```
Comment (armiol) [inline flogger/flogger-log4j2-backend/src/main/java/com/google/common/flogger/backend/log4j2/Log4j2BackendFactory.java:27]: Is it still "Flogger's"?

File: logging-backend/src/main/java/io/spine/logging/backend/system/AbstractLogRecord.java [hunk 2]
```diff
@@ -1,24 +1,30 @@
 /*
- * Copyright (C) 2015 The Flogger Authors.
+ * Copyright 2023, TeamDev. All rights reserved.
```
Comment (armiol) [inline logging-backend/src/main/java/io/spine/logging/backend/system/AbstractLogRecord.java:2]: It's important.

```
Copyright 2015, The Flogger Authors; 2023, TeamDev. All rights reserved.
```

Let's apply to all similar cases.

File: logging-backend/src/main/java/io/spine/logging/backend/system/package-info.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * Contains implementation of the default backend, which uses
+ * `java.util.logging` (JUL) to output log statements.
```
Comment (armiol) [inline logging-backend/src/main/java/io/spine/logging/backend/system/package-info.java:29]: I would use `{@code ...}` in Javadoc.

File: logging-backend/src/main/kotlin/io/spine/logging/backend/system/StdLoggerBackend.kt [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+
+    override fun handleError(error: RuntimeException, badData: LogData): Unit = doLog(
+        SimpleLogRecord.error(error, badData, Platform.getInjectedMetadata()),
+        badData.wasForced()
+    )
+
+    /**
+     * Logs the given record using the [logger] associated with this backend.
+     *
+     * If [wasForced] is set, forces the publishing by increasing the level
```
Comment (armiol) [inline logging-backend/src/main/kotlin/io/spine/logging/backend/system/StdLoggerBackend.kt:66]:
```
... is `true`...
```

File: logging/build.gradle.kts [hunk 5]
```diff
@@ -79,6 +79,8 @@ kotlin {
         }
         val jvmMain by getting {
             dependencies {
+
+                // `FluentLogger2` is exposed by the deprecated `Logging` interface.
```
Comment (armiol) [inline logging/build.gradle.kts:83]: Please add more context details here. It's difficult to comprehend without being deep into the problem.

File: tests/jvm-log4j-backend-our-context/build.gradle.kts [hunk 6]
```diff
@@ -31,6 +31,12 @@ plugins {
 dependencies {
     testImplementation(project(":logging"))
     testImplementation(project(":fixtures"))
+
+    /**
+     * Backend and context should have been added to [runtimeOnly].
+     * But tests make sure that the platform indeed picked up the expected
```
Comment (armiol) [inline tests/jvm-log4j-backend-our-context/build.gradle.kts:37]: Again, to my understanding, a few steps are skipped in this explanation. Please expand.

File: tests/jvm-our-backend-our-context/build.gradle.kts [hunk 7]
```diff
@@ -31,6 +31,12 @@ plugins {
 dependencies {
     testImplementation(project(":logging"))
     testImplementation(project(":fixtures"))
-    testRuntimeOnly(project(":logging-backend"))
-    testRuntimeOnly(project(":logging-context"))
+
+    /**
+     * Backend and context should have been added to [runtimeOnly].
```
Comment (armiol) [inline tests/jvm-our-backend-our-context/build.gradle.kts:36]: Same as above.

**Review by armiol — CHANGES_REQUESTED** [review]
@yevhenii-nadtochii please see my comments.

And yes, I would create issues on top of your "summary" findings.

File: logging-backend/src/main/java/io/spine/logging/backend/system/AbstractBackend.java [hunk 8]
```diff
@@ -26,6 +37,9 @@
 /**
  * Common backend to handle everything except formatting of log message and metadata. This is an
  * unstable implementation and should not be used outside of the Flogger core library.
+ *
+ * @see <a href="https://github.com/google/flogger/blob/cb9e836a897d36a78309ee8badf5cad4e6a2d3d8/api/src/main/java/com/google/common/flogger/backend/system/AbstractBackend.java">
```
Comment (alexander-yevsyukov) [inline logging-backend/src/main/java/io/spine/logging/backend/system/AbstractBackend.java:41]: Just a thought. We probably need to consider using some URL shortener some day. There was free one from Google, but it's no longer available. I couldn't find a good alternative.

**Review by alexander-yevsyukov — APPROVED** [review]
LGTM

File: logging-backend/src/main/java/io/spine/logging/backend/system/AbstractBackend.java (hunk 8 above)
Comment (alexander-yevsyukov) [reply logging-backend/src/main/java/io/spine/logging/backend/system/AbstractBackend.java:41]: @armiol, here's one option, which is free and looks as usable. Please have a look.

Comment (alexander-yevsyukov) [reply logging-backend/src/main/java/io/spine/logging/backend/system/AbstractBackend.java:41]: https://free-url-shortener.rb.gy/

**Comment by yevhenii-nadtochii** [comment]
@armiol PTAL

**Review by armiol — APPROVED** [review]

**Pull request merged**